src/GeoExt/tree/LayerTreeBuilder.js

```diff
@@ -0,0 +1,261 @@
+                    // the children and vice versa
+                    var me = this;
+                    if(node.data.checked) {
+                        node.eachChild(function(child) {
+                            child.set('checked',true);
+                            me.fireEvent('checkchange',child,true);
+                        });
+                    } else {
+                        node.eachChild(function(child) {
+                            child.set('checked',false);
+                            me.fireEvent('checkchange',child,false);
+                        });
+                    }
+                    this.updateCheckboxes(node.parentNode);
+                },
+                scope: this
+            });
+        }
+    },
+
+    // private
+    processLayerStore: function() {
+        if (this.layerStore.getCount() > 0) {
+            this.onLayerAdded(
+                this.layerStore,
+                this.layerStore.data.items
+            );
+        }
+    },
+
+
+    // make sure that a parent checkbox is only checked if all of it's
+    // children are also checked
+    updateCheckboxes: function(node) {
+        if(node.isRoot()) return;
+        var allChecked = true;
+        for(var i=0; i<node.childNodes.length; i++) {
+            if(!node.childNodes[i].data.checked) allChecked = false;
+        }
+        // check according to checkableContainerGroupNodes and checkableLeafGroupNodes properties
+        if (node.plugins && node.plugins[0] && node.plugins[0].pype == "gx_layergroupcontainer") {
+            checkableNode = this.checkableLeafGroupNodes;
+        } else {
+            checkableNode = this.checkableContainerGroupNodes && this.checkableLeafGroupNodes;
+        }
+        if (checkableNode) {
+            node.set('checked',allChecked);
+        }
+        this.updateCheckboxes(node.parentNode);
+    },
+
+    onLayerAdded: function(store, records, index) {
+        // first, validate all 'group' options
+        Ext.each(records, function(record, index) {
+            var layer = record.getLayer();
+
+            if(layer.displayInLayerSwitcher === false) {
+                if(layer.group && layer.options && layer.options.group) {
+                    delete layer.group;
+                    delete layer.options.group;
+                }
+                return;
+            } else if(layer.options && layer.options.group === undefined) {
+                layer.options.group = (layer.isBaseLayer)
+                    ? this.baseLayersText : this.otherLayersText;
+            }
+        }, this);
+
+        // then, create the nodes according to the records
+        Ext.each(records, function(record, index) {
+            var layer = record.getLayer();
+
+            if (layer.displayInLayerSwitcher === false) {
+                return;
+            }
+
+            var groupString = layer.options.group || "";
+            var group = groupString.split('/');
```

Comment:
Shouldn't this happen in after it has been ensured ``groupString !== ""`` in line 175?